### PR TITLE
[SeaTunnel #1044][Bug][Config] Fix quoted string in config key

### LINKED ISSUE #1044: [Bug] [Config] The key of config includes ", so that setting directly to sparkconf does not take effect
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

The key of config includes ", so that setting directly to sparkconf does not take effect

image

### SeaTunnel Version

2.0.5-SNAPSHOT

### SeaTunnel Config

```conf
env {
  spark.master = "local"
  spark.app.name = "seatunnel"
  spark.cassandra.connection.host = "127.0.0.1"
}

source {
  Cassandra {
    table.name = "t2"
    keyspace = "excelsior"
    result_table_name = "cassandra"
  }
}


transform {

}

sink {
  console {}
}
```

### Running Command

```shell
.
```

### Error Exception

```log
.
```

### Flink or Spark Version

spark 2.4.0

### Java or Scala Version

jdk8

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by simon824**
## Purpose of this pull request
closes #1044

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-config/src/main/java/com/typesafe/config/impl/Path.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    }
+
+    private void appendToStringBuilder(StringBuilder sb) {
+        if (hasFunkyChars(first) || first.isEmpty()) {
+            sb.append(ConfigImplUtil.renderJsonString(first));
+        } else {
+            sb.append(first);
+        }
+        if (remainder != null) {
+            sb.append(ConfigParseOptions.PATH_TOKEN_SEPARATOR);
```
Comment (simon824) [inline seatunnel-config/src/main/java/com/typesafe/config/impl/Path.java:224]: cc @leo65535  FYI https://github.com/apache/incubator-seatunnel/blob/v2.0.4/waterdrop-config/src/main/java/io/github/interestinglab/waterdrop/config/impl/Path.java#L203

**Comment by simon824** [comment]
PTAL @CalvinKirs @zixi0825 @leo65535  ,Thx.

**Comment by leo65535** [comment]
hi @simon824, please use following command to rerun the workflow.
```
git commit --allow-empty -m "trigger ci"
git push
```

File: seatunnel-config/src/main/java/com/typesafe/config/impl/Path.java (hunk 1 above)
Comment (xleoken) [reply seatunnel-config/src/main/java/com/typesafe/config/impl/Path.java:224]: thanks, will check it.

**Review by xleoken — APPROVED** [review]
LGTM, thanks.

**Pull request merged**